### ISSUE TITLE
Filter based on contact groups of account

### ISSUE THREAD

**Issue by dschuermann** [SufficientlySecure/birthday-calendar#59]
_Original author: domschuermann@gmail.com (June 01, 2013 10:52:00)_

Add option to filter based on group of account?

_Original issue: http://code.google.com/p/birthday-adapter/issues/detail?id=57_

**Comment by tribut** [comment]
That would really be helpful, my calendar is getting crowded :)

**Comment by blausand** [comment]
+1. A close thought!

**Comment by Otiel** [comment]
I would also love to see that.

**Comment by call-me-matt** [comment]
Filtering based on groups is implemented and will come with the next release.

I chose to filter a contact only if all blacklisted groups match. If a contact is a friend and a colleague and you blacklist colleagues, the contact will still be considered.

**Comment by Otiel** [comment]
Thank you!  I will test as soon as the F-Droid release is available. 😀

**Comment by Otiel** [comment]
The group filtering doesn't wok for me (Birthday Adapter v3.4): I still have all birthdays on my calendar, even though I just enabled "Group Filtering" and selected only one group which contains only one birthday:

<img width="1080" height="546" alt="Image" src="https://github.com/user-attachments/assets/1e47655f-6050-485b-b271-38e87ad694af" />

<img width="1080" height="1519" alt="Image" src="https://github.com/user-attachments/assets/2d77ccb5-4e6e-4bda-8b79-4ead9feda3da" />

I've tried multiple times to synchronize the calendar, and event to disable/enable "Birthday calendar" from the "Main" tab.

Let me know if I can provide some debug info or help in any other way.

**Comment by Otiel** [comment]
I'm getting this error when I try to update with the v3.5beta4 😕 

<img width="1035" height="463" alt="Image" src="https://github.com/user-attachments/assets/f4cc59b0-4297-4cab-80ba-1fdcc7aefc7a" />

Do you have an idea why?

I'm a bit lazy to uninstall the v3.4 right now because I don't want to lose all my preferences (particularly my custom labels). Do you know if I could backup my preferences by any chance (I could not find any corresponding file inside `Android/data/`)? If so, I could try to uninstall v3.4 in order to try and install v3.5beta4.

**Comment by call-me-matt** [comment]
Yes, because it uses the same app name. I do not dare to change it on my side, so let's see if it works with the next version.

Just to be sure: the contact you want to filter is only in the "Birthday notification" group and not in "ICE", right?

**Comment by Otiel** [comment]
> Just to be sure: the contact you want to filter is only in the "Birthday notification" group and not in "ICE", right?

Yes, that is right.

**Comment by Otiel** [comment]
I made some effort 😄. Uninstalled the v3.4 then installed the v3.5beta2.0415a40.

* Cutoff button is fixed ✅ 
* It indicates "last sync:  0 minutes ago" ✅ 
* But the filtering still doesn't work, all my birthdays are displayed on the calendar

**Comment by Otiel** [comment]
I'm using Samsung Calendar, not Fossify. Any change I make to the preferences (color, custom labels) is immediately visible on the Calendar after syncing. Also, I'm using Samsung Contacts app, not sure if it's of importance.

I've made more tests and I found out the following with this groups configuration:

<img width="478" height="445" alt="Image" src="https://github.com/user-attachments/assets/f907ca4c-a7eb-4d85-9e7e-257a07502550" />

* Selecting only "No group" works correctly: birthdays of contacts in groups "Birthday notification" and "ICE" are not visible anymore on the Calendar. ✅ 
* Selecting only "ICE" displays "ICE" contacts ✅ and hides "Birthday notification" contacts ✅ but "No group" contacts are still visible ❌ 

⇒ "No group" contacts are always displayed no matter its selected state, whereas the selected state is correctly applied for actual groups.

**Comment by MikevanDongen** [comment]
Hi @call-me-matt, is there any chance the last issue mentioned in the comments has resurfaced? I'm experiencing the exact situation @Otiel had described: contacts without any group are always displayed, regardless of the "No Group" checkbox. I'm on the latest version, 3.13.4, installed through F-Droid.

**Comment by call-me-matt** [comment]
You are right. When the "no group" is translated into other languages, the filter is not working. I will fix that. Thank you for bringing it up 👍

**Comment by MikevanDongen** [comment]
Thank you, @call-me-matt! I can confirm it's fixed in v3.13.7.
